packages/rum-core/src/domain/error/errorCollection.ts

```diff
@@ -113,3 +115,10 @@ function processError(error: RawError, pageStateHistory: PageStateHistory): RawR
     domainContext,
   }
 }
+
+function tryToGetErrorContext(originalError: unknown) {
+  if (isError(originalError) && 'dd_context' in originalError) {
```

Comment:
should we check that it's an `Error`? Currently we allow users to throw just about anything

to allow for more flexibility we really only need for this to be an object

https://www.typescriptlang.org/play/?#code/CYUwxgNghgTiAEYD2A7AzgF3kmBLA5rilBAKIww4Bc8ArigNYpIDuKA3AFCfLpYAWUNABFhAYVQYQADywBeeAAo0MMDXpNWKAJQ0VYeLjTwA3vGDAA+rymz1jZm3gBfeHIB8neN-gYAngAOIEgAZvD6bnIKAORIAEYAVuAY0fAAZGnhqvAAhAootBAQ6ZnRFtaSMimGKFlgXNyQQsZitJhIALbklDDwVSAowMbdOKZePvggWOU2VYraYz5L8HAYtDC1ZgwgfjTRAG4ktCCpzlxLzpyXPKhoSBAgAHQQSPiK9KAhRCDAADTwghE4kqsnegxAXxQP202i4vDuD2er0UBSK-0BogkKFsGBRhQgMLht3uTxeb2imDwKHw0XRQkxINxFIwVJphJu6BJSPJBQ6cRAMFpAPpwOxcwADOz4VyyYpomA2hhOn0KDghRjRTiUSAWPBWu0uqqYPMpcTEbLYolkuqRVitWYZoyaCZXM5CUA